ecosystem/typescript/sdk/SPEC.md

```diff
@@ -0,0 +1,210 @@
+
+### 3. Account Management
+
+- [ ] **3.1** The SDK **must** support account creation
+- [ ] **3.2** The SDK **must** support account balance management
+- [ ] **3.3** The SDK **must** support key management for Ed25519 keys
+- [ ] **3.4** The SDK **must** support local single-threaded sequence number management
+- [ ] **3.5** The SDK **should** provide multi-agent signer support
+- [ ] **3.6** The SDK **should** provide multi-Ed25519 signer support
+- [ ] **3.7** The SDK **should** provide mnemonic support
+- [ ] **3.8** The SDK **should** provide multi-sig support
+- [ ] **3.9** The SDK **should** provide resource account support
+
+### 4. Transaction
+
+- [ ] **4.1** The SDK **must** support entry function payload transaction submission
+- [ ] **4.2** The SDK **must** support script payload transaction submission
+- [ ] **4.3** The SDK **must** support gas estimation based on API calls
+- [ ] **4.4** The SDK **must** support simulation of transactions
+
+### 5. Coin Management
+
+- [ ] **5.1** The SDK **must** support APT coin transfer
+- [ ] **5.1** The SDK **should** support other coin transfer
+
+## Maintenance Requirements
+
+### 6. Source Control
+
+- [ ] **6.1** The source code for the SDK **must** be maintained within Git version control
+- [ ] **6.2** The source code **must** be hosted publicly
+- [ ] **6.3** Development of new features **should** happen on feature branches
+- [ ] **6.4** Feature branches **should** pass all tests and linting before they can be merged into the `main` branch
+- [ ] **6.5** Source control **should** contain tags for each release of the SDK
+- [ ] **6.6** The `main` branch **should** be kept in a condition that allows for direct use through checkout
+- [ ] **6.7** The source code **should** use GitHub for public hosting
+
+### 7. Releases & Versioning
+
+- [ ] **7.1** The SDK **must** use [Semantic Versioning](http://semver.org/) to increment the version number as changes are made
+- [ ] **7.2** For every new release the `CHANGELOG` file **must** to be updated with the `Major`, `Minor` and `Patch` changes
+- [ ] **7.3** A release package **must** include the documentation `README` file
+- [ ] **7.4** A release package **must** include the `LICENSE` file
+- [ ] **7.5** A release package **must** include the `CHANGELOG` file
+
+- [ ] **7.6** A release package **should** not include unnecessary source code files or intermediarry files for the SDK.
+
+- [ ] **7.7** The name of the SDK **should** follow language best practices, and be one of `aptos` or `Aptos`
+- [ ] **7.8** If the preferred name of the SDK is not available, it **could** be one of `aptos-sdk`, `AptosSDK`, or `aptosdev`.
+- [ ] **7.9** As soon as the first public version of the library has been signed off, the version **should** be bumped to `1.0.0`
+- [ ] **7.10** New releases **could** be deployed automatically to the package manager using the CI server
+
+### 8. CI Server
+
```

Comment:
test shouldn't run against mainnet - we dont want to submit transaction to mainnet.
currently, test run against local testnet (which make sense) and indexer tests (or any tests involved indexer) run against testnet - that's because indexer on devnet is not reliable.

In my mind, we should run tests against local testnet that includes full node and indexer.